### PR TITLE
[docs] Introduce custom Jekyll tags for better glossary integration

### PR DESCRIPTION
**Pull request by abiogenesis-now**
**Some Context**:
Jekyll supports integrations like custom tags (e.g. think `{% include %}`, but with behavior that you define yourself). @heckj is working on the user journey and is need of an easy way to bring in the [glossary work](https://github.com/kubernetes/website/pull/5657).

**TLDR**:
This PR implements tags to help you incorporate glossary definitions when needed. For the full spec, see the [`README`](https://github.com/abiogenesis-now/kubernetes.github.io/blob/jyao/jekyll-tags-for-glossary/_plugins/README.md) that I added in the `_plugins/` directory.

CC @chenopis and @bradamant3 who are actively working on user journeys as well

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/5954)

### CODE REVIEW

**Comment by abiogenesis-now** [comment]
oh additional note worth mentioning: I had to disable the Github Pages gem in favor of Jekyll gem(s) because the former doesn't support custom plugins (only if they're from "core").

Locally, the site seemed to still render alright---and since we're hosting on Netlify, not GH pages, I *think* the update should be ok. Will double check once the deploy preview is out.

**Comment by zacharysarah** [comment]
@abiogenesis-now 👋 If you tell me what to look for on the staging server, I'm happy to verify it.

**Comment by abiogenesis-now** [comment]
@zacharysarah Thanks! So the new tags are being used in two places:

1. In the brief summary of StatefulSets here: https://deploy-preview-5954--kubernetes-io-master-staging.netlify.com/docs/concepts/workloads/controllers/statefulset/

2. As a tooltip for Pod here: https://deploy-preview-5954--kubernetes-io-master-staging.netlify.com/docs/tasks/inject-data-application/define-command-argument-container/

If those look ok to you we should be good to go I think.

**Comment by zacharysarah** [comment]
@abiogenesis-now After testing, it looks like the tags on the [first page](https://deploy-preview-5954--kubernetes-io-master-staging.netlify.com/docs/concepts/workloads/controllers/statefulset/) don't render correctly, while the tag on the [second page](https://deploy-preview-5954--kubernetes-io-master-staging.netlify.com/docs/tasks/inject-data-application/define-command-argument-container/) works.

First page (glossary links don't work):
screen shot 2017-10-19 at 12 29 35 pm

screen shot 2017-10-19 at 12 33 49 pm

Second page (glossary link works):

screen shot 2017-10-19 at 12 29 11 pm

**Comment by abiogenesis-now** [comment]
ahh @zacharysarah good catch! that is very important 😅 

will fix and ping you when updated

**Comment by abiogenesis-now** [comment]
@zacharysarah @chenopis when either of you have a chance, do you mind taking a look at this PR again? I know it's rather long, but I've fixed the link issues that Zach pointed out with the previous iteration. Thank you!!

*TLDR*: Now if you use the `glossary_tooltip` tag, you get a tooltip AND a link, which either:
* goes to the `full-link` specified in the YAML file of that glossary term
* falls back to the glossary entry

*Screenshotted below*:

screen shot 2017-11-01 at 1 29 52 pm

**Comment by chenopis** [comment]
@zacharysarah @steveperry-53 PTAL when you get a chance also. I'm not sure who can do the tech review though.

**Comment by zacharysarah** [comment]
> Now if you use the glossary_tooltip tag, you get a tooltip AND a link

🎉 I like it! I'm not sure about tech review, either&mdash;but that seems like the sort of thing we should have in our pocket. Maybe @ahmetb? (Hi there!)

**Comment by ahmetb** [comment]
Hi, I haven't been following the space and am very much far away from what’s been happening in Glossary work. I do not carry any Jekyll expertise either, so I cannot be very helpful. I’m sorry.

**Comment by chenopis** [comment]
@zacharysarah Well, this looks like pretty standard plugin code. The JS is pretty short and straightforward. Unless you see anything buggy in the preview, I think we can go ahead and merge it. I tested it as best as I could.

**Comment by abiogenesis-now** [comment]
Thanks for the review everyone! Just let me know if you have any remaining questions!

**Comment by zacharysarah** [comment]
@chenopis Agreed, nothing I’m seeing raises concern. @abiogenesis-now :shipit: :tada:

**Comment by zacharysarah** [comment]
@ahmetb Thanks for responding, no worries!
